src/test_lib_json/main.cpp

```diff
@@ -308,6 +311,72 @@ JSONTEST_FIXTURE(ValueTest, arrayIssue252) {
   // JSONTEST_ASSERT_EQUAL(5, root["array"].size());
 }
 
+JSONTEST_FIXTURE(ValueTest, arrayInsertAtRandomIndex) {
+  Json::Value array;
+  Json::Value str0("index2");
+  Json::Value str1("index3");
+  array.append("index0"); // append rvalue
+  array.append("index1");
+  array.append(str0); // append lvalue
+
+  std::vector<Json::Value*> vec; // storage value address for checking
+  for (int i = 0; i < 3; i++) {
```

Comment:
Why have a temporary array at all? It's not clear to me what taking the address of each array index and checking them against themselves does. Are you testing that the array accessor doesn't mutate the list?

Reply:
Yes.
It is used to test the array accessor doesn't mutate the list after inserting a value and used to make sure that the key-value pair in value.map_ had not be changed.

Reply:
chang to return false only